src/controller/src/lib.rs

```diff
@@ -693,6 +717,12 @@ where
             unfulfilled_watch_sets: BTreeMap::new(),
             watch_set_id_gen: Gen::default(),
             immediate_watch_sets: Vec::new(),
+        };
+
+        if !this.read_only {
+            this.remove_past_generation_replicas_in_background();
```

Comment:
Seems a little weird to have this call in two places. Can we just call `allow_writes` here instead?

Reply:
The issue is that `allow_writes` as currently written will just return doing nothing. Maybe we can just remove the `compute.read_only()` guard and make all calls in `allow_writes` idempotent? Actually I think they already are. Could also keep the check but initialize the global controller with `read_only: false` above.

Reply:
Good catch! I went with what Jan suggested: initialize `Self::read_only` to `false`, then conditionally call `allow_writes()`.